src/TimeSeries/widget/TimeSeries.ts

```
@@ -0,0 +1,191 @@
+        } else {
+            logger.error(this.id + ".updateData unknown source or error in widget configuration");
+            callback();
+        }
+    }
+
+    /**
+     * Validate the widget configurations from the modeler
+     */
+    private checkConfig() {
+        let valid = true;
+        const incorrectSeries = this.seriesConfig.filter(series =>
+            (series.seriesSource === "microflow" && !series.dataSourceMicroflow)
+        );
+
+        if (incorrectSeries.length) {
+            valid = false;
+            mx.ui.error("Configuration error for series : '" + incorrectSeries[0].seriesKey +
+                "'. Source is set to 'Microflow' but 'Source - microflow' is missing ", true);
+        }
+
+        return valid;
+    }
+
+    private updateRendering (callback?: Function) {
+        render(createElement(TimeSeries, this.createProps()), this.domNode);
+
+        mxLang.nullExec(callback);
+    }
+
+    private resetSubscriptions () {
+        mx.data.unsubscribe(this.handle);
+        this.handle = 0;
+
+        if (this.contextObject) {
+            this.handle = this.subscribe({
+                callback: (guid: string) => {
+                    this.updateRendering();
+                },
+                guid: this.contextObject.getGuid()
+            });
+        }
+    }
+
+    private fetchDataFromXpath(seriesConfig: SeriesConfig, callback: Function) {
+        if (this.contextObject) {
+            const guid = this.contextObject ? this.contextObject.getGuid() : "";
+            const constraint = seriesConfig.entityConstraint.replace("[%CurrentObject%]", guid);
+            const xpathString = "//" + seriesConfig.seriesEntity + constraint;
+            mx.data.get({
+                callback: callback.bind(this),
+                error: (error) => {
+                    logger.error(this.id + ": An error occurred while retrieving data: ", error);
+                    mx.ui.error("An error occurred while retrieving data");
```

Ah yet another way of handling errors :-) Now logging and calling `mx.ui.error`. Can't you let the errors bubble up and handle them in one place, to avoid having error handling all of the place.